lib/sles4sap/sap_deployment_automation_framework/basetest.pm

```diff
@@ -95,6 +96,32 @@ sub full_cleanup {
 
 sub post_fail_hook {
     record_info('Post fail', 'Executing post fail hook');
+    if (testapi::is_serial_terminal()) {
+        if ($serial_regexp_playbook) {
+            # Teminate the process when playbook timed out
+            # Note: please do not use '> ' (or $testapi::distri->{serial_term_prompt})
+            # as there are lots of '> ' in the oupput of playbook
+            unless (wait_serial(qr/-\d+-/)) {
+                type_string('', terminate_with => 'ETX');
+                # Wait 5s for process returns and serial terminal gets ready
+                sleep 5;
+                type_string("\n");
+                record_info('Terminate playbook process');
+            }
+            $serial_regexp_playbook = 0;
+        }
+        else {
+            # Teminate the process when other scripts timed out
+            unless (wait_serial($testapi::distri->{serial_term_prompt})) {
+                type_string('', terminate_with => 'ETX');
+                # Wait 5s for process returns and serial terminal gets ready
+                sleep 5;
```

Comment:
Same as above. Replacing sleep with wait_serial.

Reply:
See above reponse /reason for `sleep 5`.